### PR TITLE
Add hourglass architecture and some structure to the repo

### PR DESCRIPTION
**Pull request by robertklee**
Add a hourglass architecture with a variable number of stacks, variable input size, etc.

### CODE REVIEW

File: hourglass.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+                                                  self.num_channels, self.inres, self.outres, bottleneck_mobile)
+        else:
+            self.model = create_hourglass_network(self.num_classes, self.num_stacks,
+                                                  self.num_channels, self.inres, self.outres, bottleneck_block)
+        # show model summary and layer name
+        if show:
+            self.model.summary()
+        
+    def train(self, batch_size, model_path, epochs):
+        # TODO
```
Comment (julianrocha) [inline hourglass.py:36]: Assuming this is where we would interface with the input pipeline. Something like:
```python
model = # load model from model_path and compile etc...
train_gen = DataGenerator("cleaned_dataframe.csv", "data", (244, 244), batch_size=batch_size, shuffle=True)
val_gen = DataGenerator(...)
model.fit(train_gen,val_gen,epochs=epochs...)
```
[Reference](https://mahmoudyusof.github.io/facial-keypoint-detection/data-generator/)

File: hourglass_blocks.py [hunk 2]
```diff
@@ -0,0 +1,193 @@
+import keras.backend as K
+from keras.layers import *
+from keras.losses import mean_squared_error
+from keras.models import *
+from keras.optimizers import Adam, RMSprop
+
+# Adapted from https://github.com/yuanyuanli85/Stacked_Hourglass_Network_Keras/blob/master/src/net/hg_blocks.py
+
+def create_hourglass_network(num_classes, num_stacks, num_channels, inres, outres, bottleneck):
+    input = Input(shape=(inres[0], inres[1], 3))
```
Comment (julianrocha) [inline hourglass_blocks.py:10]: Configurable input size, 3 channels, any idea what dimensions would work best for this arch?

File: hourglass_blocks.py [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+                  name=str(hgid) + '_conv_1x1_x2')(head)
+    head_m = Conv2D(num_channels, kernel_size=(1, 1), activation='linear', padding='same',
+                    name=str(hgid) + '_conv_1x1_x3')(head_parts)
+
+    head_next_stage = Add()([head, head_m, prelayerfeatures])
+    return head_next_stage, head_parts
+
+
+def euclidean_loss(x, y):
+    return K.sqrt(K.sum(K.square(x - y)))
```
Comment (julianrocha) [inline hourglass_blocks.py:193]: Is this a temporary loss function, will we be replacing this?

File: scripts/coco_dl.sh [hunk 4]
```diff
@@ -0,0 +1,38 @@
+# Credit to https://gist.github.com/mkocabas/a6177fc00315403d31572e17700d7fd9
```
Comment (julianrocha) [inline scripts/coco_dl.sh:1]: nice

**Review by julianrocha — COMMENTED** [review]
Great work Robert this looks awesome. Some of the details in `hourglass_blocks.py` are a little confusing to me so wouldn't mind a brief overview in the meeting tn.

File: hourglass_blocks.py (hunk 2 above)
Comment (robertklee) [reply hourglass_blocks.py:10]: yep, these are given in `train.py`. Num classes is the number of joints, so for us it is 17. Stacks should be 2-8, with the paper we are basing it off of using 8.

File: hourglass_blocks.py (hunk 3 above)
Comment (robertklee) [reply hourglass_blocks.py:193]: Temporary for now

File: hourglass.py (hunk 1 above)
Comment (robertklee) [reply hourglass.py:36]: Yep, there's still a bit to be written here. Mostly data generator, callbacks, model saving.
